src/data-workspace/category-combo-table-body-pivoted/generate-form-matrix/generate-matrix-transposed.js

```diff
@@ -0,0 +1,77 @@
+import { selectors } from '../../../shared/index.js'
+
+export const generateMatrixTransposed = (options) => {
+    const {
+        metadata,
+        categoryOptionsDetails,
+        sortedCOCs,
+        categories,
+        dataElements,
+    } = options
+
+    const columnHeaderFields = [...categories, ...dataElements]
+
+    const columnHeaders = [
+        [
+            ...columnHeaderFields.map((header) => {
+                return {
+                    id: header?.id,
+                    displayFormName: header?.displayFormName,
+                    type: 'columnHeader',
+                    metadataType: header.valueType ? 'dataElement' : 'category',
+                }
+            }),
+        ],
+    ]
+
+    const rowHeaders = []
+
+    const alreadyAdded = {}
+    sortedCOCs.forEach((categoryOptionCombo) => {
+        const dataEntryRow = []
+
+        categoryOptionCombo.categoryOptions.forEach((fco) => {
+            const categoryOption = categoryOptionsDetails.find(
+                (cod) => cod.id === fco
+            )
+            const categoryIndex = categories.findIndex((cat) =>
+                cat.categoryOptions.includes(fco)
+            )
+
+            // todo(pivot): maybe refactor this logic after confirming this is Functional Design team wants UI-wise
+            const lastCateogry = categoryIndex === categories.length - 1
+            if (!alreadyAdded[categoryOption.id] || lastCateogry) {
+                dataEntryRow.push({
+                    id: categoryOption?.id,
+                    displayFormName: categoryOption?.displayName,
+                    rowSpan:
```

Comment:
I'm not sure this is working as intended? When I try it with [Emergency Response data set](http://localhost:3000/#/?attributeOptionComboSelection=eLwL77Z9E7R-JvIqWKLPPkt&dataSetId=Lpw6GcnTrmS&orgUnitId=ImspTQPwCqd&periodId=202402&features=pivot), I get a misformed table (due to miscalculated rowspan): 
<img width="1132" alt="image" src="https://github.com/dhis2/aggregate-data-entry-app/assets/18490902/dff323e8-d308-40df-994d-3cd1e53a39d7">

Reply:
fixed